storage/response_cache.go

```diff
@@ -246,29 +200,6 @@ func (rc *ResponseCache) shouldCacheResponse(reply proto.Response) bool {
 	return true
 }
 
-// addInflightLocked adds the supplied ClientCmdID to the inflight
-// map. Any subsequent invocations of GetResponse for the same client
-// command will block on the inflight cond var until either the
-// response cache is cleared or this command is removed via
-// PutResponse().
-func (rc *ResponseCache) addInflightLocked(cmdID proto.ClientCmdID) {
```

Comment:
Actually, we apply raft commands serially, so the first increment (say) puts its result in the response cache and all subsequent repeat requests will get a cache hit. So it's raft itself which provides the synchronization we need, not a mutex.